deployment/README.md

```diff
@@ -0,0 +1,186 @@
+
+Runs the scripts/custom.sh script. This is meant as a convenience tool for executing ad-hoc scripts.
+
+Applies to all hosts by default, unless other nodes are specified using --extra-vars "nodes=..."
+
+### `deep-clean.yaml`
+
+Performs deep cleaning of the host machines.
+Runs clean.yaml and, in addition, deletes the cloned repository with the lotus code and binaries.
+
+Applies to all hosts by default, unless other nodes are specified using --extra-vars "nodes=..."
+
+### `deploy-current.yaml`
+
+Deploys the bootstrap and the validators using existing binaries.
+This playbook still cleans the lotus daemon state, but neither updates nor recompiles the code.
+Performs the state cleanup by running clean.yaml,
+potentially producing and ignoring some errors, if nothing is running on the hosts - this is normal.
+
+The nodes variable must not be set, as this playbook must distinguish between different kinds of nodes
+(such as bootstrap and validators).
+
+### `deploy-new.yaml`
+
+Deploys the whole system from scratch.
+Performs a deep clean by running deep-clean.yaml
+(potentially producing and ignoring some errors, if nothing is running on the hosts - this is normal)
+and sets up a new Spacenet deployment.
+
+The nodes variable must not be set, as this playbook must distinguish between different kinds of nodes
+(such as bootstrap and validators).
+
+### `generate-genesis.yaml`
+
+Generates a new genesis block (at the bootstrap node) and copies it to all validators.
+Assumes the hosts to have already been set up (using setup.yaml).
+
+An alternative set of nodes to copy the genesis block to can be specified using --extra-vars "nodes=..."
+
+### `kill.yaml`
+
+Kills running Lotus daemon and Mir validator.
+Does not touch their persisted state or the code and binaries.
+Reports but ignores errors, so it can be used even if the processes to be killed are not running.
+
+Applies to all hosts by default, unless other nodes are specified using --extra-vars "nodes=..."
+
+### `restart-validators.yaml`
+
+Restarts a given set of validators.
+For safety, does NOT default to restarting all validators
+and the set of hosts to restart must be explicitly given using --extra-vars "nodes=..."
+
+Note that this playbook always affects all hosts, regardless of the value of the nodes variable.
+This is due to the necessity of reconnecting all daemons to the restarted one.
+
+### `setup.yaml`
+
+Sets up the environment for running the Lotus daemon and validator.
+This includes installing the necessary packages, fetching the Lotus code, and compiling it.
+It does not start any nodes. See start-* and deploy-*.yaml for starting the nodes.
+
+Applies to all hosts by default, unless other nodes are specified using --extra-vars "nodes=..."
+
+### `start-bootstrap.yaml`
+
+Starts the bootstrap node and downloads its identity to localhost.
+Assumes the host has been set up and the genesis block has been generated
+(using setup.yaml and generate-genesis.yaml respectively).
```

Comment:
Is `generate-genesis.yaml` triggered every time we call `start-bootstrap.yaml`, or if there is an existing genesis already it just distributed and apply that one? I ask because re-generating every time the bootstrap may lead to non-determinism and errors as the ones we've seen when restarting single nodes of the network.

Reply:
The genesis is not generated every time.